File: backend/src/main/java/com/soubhagya/pingme/service/AdminService.java
package com.soubhagya.pingme.service;

import com.soubhagya.pingme.dto.response.AdminDashboardStatsResponse;
import com.soubhagya.pingme.dto.response.AdminUserPageResponse;
import com.soubhagya.pingme.dto.response.UserResponse;

import java.util.List;

public interface AdminService {

    AdminDashboardStatsResponse getDashboardStats();

    AdminUserPageResponse getUsers(String status, String search, int page, int size);

    List<UserResponse> getPendingUsers();

    UserResponse getUserById(Long id);

    UserResponse approveUser(Long id);

    UserResponse rejectUser(Long id);

    void deleteUser(Long id);
}

File: frontend/src/services/AdminService.js
import api from "../api/axios";

const AdminService = {
  getDashboardStats() {
    return api.get("/admin/dashboard");
  },

  getUsers({ status = "ALL", search = "", page = 0, size = 10 }) {
    return api.get("/admin/users", {
      params: {
        status,
        search,
        page,
        size,
      },
    });
  },

  getSettings() {
    return api.get("/admin/settings");
},

  approveUser(id) {
    return api.put(`/admin/approve/${id}`);
  },

  rejectUser(id) {
    return api.put(`/admin/reject/${id}`);
  },

  resendActivationEmail(id) {
    return api.put(`/admin/resend-activation/${id}`);
},

  deleteUser(id) {
    return api.delete(`/admin/users/${id}`);
  },

  sendPasswordOtp(email) {
    return api.post(
        "/admin/send-password-otp",
        {
            email
        }
    );
},

changePassword(data) {
    return api.post(
        "/admin/change-password",
        data
    );
},
};

export default AdminService;
